java-checks/src/main/java/org/sonar/java/checks/ArrayHashCodeAndToStringCheck.java

```diff
@@ -41,6 +46,27 @@ protected MethodMatchers getMethodInvocationMatchers() {
   @Override
   protected void onMethodInvocationFound(MethodInvocationTree mit) {
     String methodName = mit.methodSymbol().name();
-    reportIssue(ExpressionUtils.methodName(mit), "Use \"Arrays." + methodName + "(array)\" instead.");
+    QuickFixHelper.newIssue(context)
+      .forRule(this)
+      .onTree(ExpressionUtils.methodName(mit))
+      .withMessage("Use \"Arrays." + methodName + "(array)\" instead.")
+      .withQuickFixes(() -> getQuickFix(mit))
+      .report();
   }
+
+  private static List<JavaQuickFix> getQuickFix(MethodInvocationTree tree) {
+    String methodName = ExpressionUtils.methodName(tree).name();
+    if ("toString".equals(methodName)) {
+      JavaQuickFix toStringQuickFix = JavaQuickFix.newQuickFix("Use \"Arrays.toString(array)\" instead.")
+        .addTextEdit(JavaTextEdit.replaceTree(tree, "Arrays.toString(" + tree.firstToken().text() + ")"))
```

Comment:
I feel this is not enough to build something that compiles. I see 2 issues:

1. The tree is not necessarily an identifier, which can lead to very different fixes:
```java
int[][] array = new int[42][42];
array[0].toString(); // you expect "Arrays.toString(array[0])", but would get  "Arrays.toString(array)"

this.foo.bar.myArray.toString();  // same problem, you would end up with "Arrays.toString(this)", which is wrong
```
2. If you don't import "java.util.Arrays", it won't compile. You need to check that it's already there, and if not, add it to the imports. Have a look at the [`QuickFixHelper.ImportSupplier`](https://github.com/SonarSource/sonar-java/blob/master/java-checks/src/main/java/org/sonar/java/checks/helpers/QuickFixHelper.java#L232-L246), which might help you understand how to inject such import. The rule `org.sonar.java.checks.CollectionImplementationReferencedCheck` already uses it and might help you as well.